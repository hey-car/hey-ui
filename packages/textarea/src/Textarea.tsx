import React, { ChangeEvent, FocusEvent, useCallback, useState } from 'react';
import cn from 'classnames';

import FormControl from '@heycar-uikit/form-control';

import TextareaCounter from './components/TextareaCounter';
import { TextareaProps } from './Textarea.types';

import styles from './styles/default.module.css';

const Textarea = React.forwardRef<HTMLTextAreaElement, TextareaProps>(
  (
    {
      value,
      defaultValue,
      hint,
      error,
      label,
      maxLength,
      fullWidth = false,
      resize = 'none',
      disabled,
      readOnly,
      onFocus,
      onBlur,
      onChange,
      className,
      dataTestId,
      ...restProps
    },
    ref,
  ) => {
    const textareaClassNames = cn(styles.textarea, className, {
      [styles.resizeVertical]: resize === 'vertical',
    });
    const isUncontrolled = value === undefined;
    const ariaLabel = typeof label === 'string' ? label : undefined;
    const [isFocused, setFocused] = useState(restProps.autoFocus);
    const [stateValue, setStateValue] = useState(defaultValue || '');
    const isFilled = Boolean(isUncontrolled ? stateValue : value);
    const handleTextareaFocus = useCallback(
      (event: FocusEvent<HTMLTextAreaElement>) => {
        if (!readOnly) setFocused(true);
        if (onFocus) onFocus(event);
      },
      [onFocus, readOnly],
    );
    const handleTextareaBlur = useCallback(
      (event: FocusEvent<HTMLTextAreaElement>) => {
        setFocused(false);

        if (onBlur) onBlur(event);
      },
      [onBlur],
    );
    const handleTextareaChange = useCallback(
      (event: ChangeEvent<HTMLTextAreaElement>) => {
        if (onChange) onChange(event, { value: event.target.value });
        if (isUncontrolled) setStateValue(event.target.value);
      },
      [onChange, isUncontrolled],
    );
    const getValueLength = (): number => {
      if (isUncontrolled) return stateValue.length;

      return value.length;
    };

    return (
      <FormControl
        bottomAddons={
          !!maxLength && (
            <TextareaCounter length={getValueLength()} maxLength={maxLength} />
          )
        }
        className={styles.formControl}
        disabled={disabled}
        error={error}
        filled={isFilled || isFocused}
        focused={isFocused}
        fullWidth={fullWidth}
        hint={hint}
        label={label}
      >
        <textarea
          {...restProps}
          aria-label={ariaLabel}
          className={textareaClassNames}
          data-test-id={dataTestId}
          disabled={disabled}
          maxLength={maxLength}
          onBlur={handleTextareaBlur}
          onChange={handleTextareaChange}
          onFocus={handleTextareaFocus}
          readOnly={readOnly}
          ref={ref}
          value={value}
        />
      </FormControl>
    );
  },
);

Textarea.displayName = 'Textarea';

export default Textarea;
