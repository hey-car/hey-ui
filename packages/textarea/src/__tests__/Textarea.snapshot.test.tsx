import React from 'react';
import { render } from '@testing-library/react';

import Textarea from '../Textarea';

describe('Textarea Snapshots tests', () => {
  it('should match callback methods', () => {
    expect(
      render(<Textarea onChange={jest.fn()} value="value" />),
    ).toMatchSnapshot();
  });

  it('should match label and value props', () => {
    expect(render(<Textarea label="Label" value="value" />)).toMatchSnapshot();
  });

  it('should match max length', () => {
    expect(
      render(<Textarea label="Label" maxLength={20} value="value" />),
    ).toMatchSnapshot();
  });

  it('should match disabled state', () => {
    expect(
      render(<Textarea disabled={true} label="Label" value="value" />),
    ).toMatchSnapshot();
  });

  it('should match error state', () => {
    expect(
      render(
        <Textarea
          error="This is the placeholder for error note"
          label="Label"
          value="value"
        />,
      ),
    ).toMatchSnapshot();
  });

  it('should match hint text', () => {
    expect(
      render(
        <Textarea
          hint="This is the placeholder for error note"
          label="Label"
          value="value"
        />,
      ),
    ).toMatchSnapshot();
  });

  it('should match fullWidth', () => {
    expect(
      render(<Textarea fullWidth={true} label="Label" value="value" />),
    ).toMatchSnapshot();
  });
});
