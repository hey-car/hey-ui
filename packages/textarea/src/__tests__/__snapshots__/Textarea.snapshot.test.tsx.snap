// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea Snapshots tests should match callback methods 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component formControl"
      >
        <div
          class="inner filled"
        >
          <div
            class="inputWrapper"
          >
            <div
              class="input"
            >
              <textarea
                class="textarea"
              >
                value
              </textarea>
            </div>
          </div>
          <fieldset
            aria-hidden="true"
            class="fieldset"
          >
            <legend
              class="legend"
            />
          </fieldset>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component formControl"
    >
      <div
        class="inner filled"
      >
        <div
          class="inputWrapper"
        >
          <div
            class="input"
          >
            <textarea
              class="textarea"
            >
              value
            </textarea>
          </div>
        </div>
        <fieldset
          aria-hidden="true"
          class="fieldset"
        >
          <legend
            class="legend"
          />
        </fieldset>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Textarea Snapshots tests should match disabled state 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component formControl disabled"
      >
        <div
          class="inner filled"
        >
          <div
            class="inputWrapper"
          >
            <label
              class="label"
            >
              <span
                class="labelInner"
              >
                Label
              </span>
            </label>
            <div
              class="input"
            >
              <textarea
                aria-label="Label"
                class="textarea"
                disabled=""
              >
                value
              </textarea>
            </div>
          </div>
          <fieldset
            aria-hidden="true"
            class="fieldset"
          >
            <legend
              class="legend"
            >
              <span>
                Label
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component formControl disabled"
    >
      <div
        class="inner filled"
      >
        <div
          class="inputWrapper"
        >
          <label
            class="label"
          >
            <span
              class="labelInner"
            >
              Label
            </span>
          </label>
          <div
            class="input"
          >
            <textarea
              aria-label="Label"
              class="textarea"
              disabled=""
            >
              value
            </textarea>
          </div>
        </div>
        <fieldset
          aria-hidden="true"
          class="fieldset"
        >
          <legend
            class="legend"
          >
            <span>
              Label
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Textarea Snapshots tests should match error state 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component formControl hasError"
      >
        <div
          class="inner filled"
        >
          <div
            class="inputWrapper"
          >
            <label
              class="label"
            >
              <span
                class="labelInner"
              >
                Label
              </span>
            </label>
            <div
              class="input"
            >
              <textarea
                aria-label="Label"
                class="textarea"
              >
                value
              </textarea>
            </div>
          </div>
          <fieldset
            aria-hidden="true"
            class="fieldset"
          >
            <legend
              class="legend"
            >
              <span>
                Label
              </span>
            </legend>
          </fieldset>
        </div>
        <span
          class="sub error"
          role="alert"
        >
          This is the placeholder for error note
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component formControl hasError"
    >
      <div
        class="inner filled"
      >
        <div
          class="inputWrapper"
        >
          <label
            class="label"
          >
            <span
              class="labelInner"
            >
              Label
            </span>
          </label>
          <div
            class="input"
          >
            <textarea
              aria-label="Label"
              class="textarea"
            >
              value
            </textarea>
          </div>
        </div>
        <fieldset
          aria-hidden="true"
          class="fieldset"
        >
          <legend
            class="legend"
          >
            <span>
              Label
            </span>
          </legend>
        </fieldset>
      </div>
      <span
        class="sub error"
        role="alert"
      >
        This is the placeholder for error note
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Textarea Snapshots tests should match fullWidth 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component formControl fullWidth"
      >
        <div
          class="inner filled"
        >
          <div
            class="inputWrapper"
          >
            <label
              class="label"
            >
              <span
                class="labelInner"
              >
                Label
              </span>
            </label>
            <div
              class="input"
            >
              <textarea
                aria-label="Label"
                class="textarea"
              >
                value
              </textarea>
            </div>
          </div>
          <fieldset
            aria-hidden="true"
            class="fieldset"
          >
            <legend
              class="legend"
            >
              <span>
                Label
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component formControl fullWidth"
    >
      <div
        class="inner filled"
      >
        <div
          class="inputWrapper"
        >
          <label
            class="label"
          >
            <span
              class="labelInner"
            >
              Label
            </span>
          </label>
          <div
            class="input"
          >
            <textarea
              aria-label="Label"
              class="textarea"
            >
              value
            </textarea>
          </div>
        </div>
        <fieldset
          aria-hidden="true"
          class="fieldset"
        >
          <legend
            class="legend"
          >
            <span>
              Label
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Textarea Snapshots tests should match hint text 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component formControl"
      >
        <div
          class="inner filled"
        >
          <div
            class="inputWrapper"
          >
            <label
              class="label"
            >
              <span
                class="labelInner"
              >
                Label
              </span>
            </label>
            <div
              class="input"
            >
              <textarea
                aria-label="Label"
                class="textarea"
              >
                value
              </textarea>
            </div>
          </div>
          <fieldset
            aria-hidden="true"
            class="fieldset"
          >
            <legend
              class="legend"
            >
              <span>
                Label
              </span>
            </legend>
          </fieldset>
        </div>
        <span
          class="sub hint"
        >
          This is the placeholder for error note
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component formControl"
    >
      <div
        class="inner filled"
      >
        <div
          class="inputWrapper"
        >
          <label
            class="label"
          >
            <span
              class="labelInner"
            >
              Label
            </span>
          </label>
          <div
            class="input"
          >
            <textarea
              aria-label="Label"
              class="textarea"
            >
              value
            </textarea>
          </div>
        </div>
        <fieldset
          aria-hidden="true"
          class="fieldset"
        >
          <legend
            class="legend"
          >
            <span>
              Label
            </span>
          </legend>
        </fieldset>
      </div>
      <span
        class="sub hint"
      >
        This is the placeholder for error note
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Textarea Snapshots tests should match label and value props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component formControl"
      >
        <div
          class="inner filled"
        >
          <div
            class="inputWrapper"
          >
            <label
              class="label"
            >
              <span
                class="labelInner"
              >
                Label
              </span>
            </label>
            <div
              class="input"
            >
              <textarea
                aria-label="Label"
                class="textarea"
              >
                value
              </textarea>
            </div>
          </div>
          <fieldset
            aria-hidden="true"
            class="fieldset"
          >
            <legend
              class="legend"
            >
              <span>
                Label
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component formControl"
    >
      <div
        class="inner filled"
      >
        <div
          class="inputWrapper"
        >
          <label
            class="label"
          >
            <span
              class="labelInner"
            >
              Label
            </span>
          </label>
          <div
            class="input"
          >
            <textarea
              aria-label="Label"
              class="textarea"
            >
              value
            </textarea>
          </div>
        </div>
        <fieldset
          aria-hidden="true"
          class="fieldset"
        >
          <legend
            class="legend"
          >
            <span>
              Label
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Textarea Snapshots tests should match max length 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component formControl"
      >
        <div
          class="inner filled"
        >
          <div
            class="inputWrapper"
          >
            <label
              class="label"
            >
              <span
                class="labelInner"
              >
                Label
              </span>
            </label>
            <div
              class="input"
            >
              <textarea
                aria-label="Label"
                class="textarea"
                maxlength="20"
              >
                value
              </textarea>
            </div>
          </div>
          <fieldset
            aria-hidden="true"
            class="fieldset"
          >
            <legend
              class="legend"
            >
              <span>
                Label
              </span>
            </legend>
          </fieldset>
        </div>
        <div
          class="counter"
        >
          <div
            class="countLength"
          >
            5
          </div>
          /
          20
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component formControl"
    >
      <div
        class="inner filled"
      >
        <div
          class="inputWrapper"
        >
          <label
            class="label"
          >
            <span
              class="labelInner"
            >
              Label
            </span>
          </label>
          <div
            class="input"
          >
            <textarea
              aria-label="Label"
              class="textarea"
              maxlength="20"
            >
              value
            </textarea>
          </div>
        </div>
        <fieldset
          aria-hidden="true"
          class="fieldset"
        >
          <legend
            class="legend"
          >
            <span>
              Label
            </span>
          </legend>
        </fieldset>
      </div>
      <div
        class="counter"
      >
        <div
          class="countLength"
        >
          5
        </div>
        /
        20
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
